daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
I agree that network and driver options are not to be mixed. As far as I can tell, [they aren't on libnetwork side](https://github.com/chenchun/libnetwork/blob/706968655f9281857ae24e01503422200dea208a/api/types.go#L39-L40), but it's not obvious from the caller code.